src/Context.hs

```diff
@@ -12,25 +12,13 @@ module Context (
     pkgConfFile, objectPath
     ) where
 
-import GHC.Generics
+import Context.Type
```

Comment:
As an example of the kind of comment I'm looking for:
```
-- The module "Context.Type" is used to break import cycles caused by "Oracles.Setting".
import Context.Type
```

Reply:
I doubt those comments will be of much use. They will document what the initial cycle that was broken was, but will ultimately bitrot pretty quickly.  Any further refactoring or addition of code to hadrian that uses the `.Type` modules will not see that there could have been a cycle without the `.Type` module and subsequently not update the comment.

As such I believe the comment while it might be initially correct will be ultimately incorrect.

Reply:
Yes, I agree, these comments will likely get out-of-date quickly, and we probably don't want to keep changing the structure very often.

Instead it may be more useful to add a general comment somewhere in the overview document that explains the module structure. (We don't have such a document at the moment, so we don't need to do it now.)

Reply:
I suggest we write such a description once we have integrated all of #445 as things will be in flux until we're done with that.